llvm/lib/MCCAS/MCCASObjectV1.cpp

```diff
@@ -2846,25 +2855,32 @@ MCCASReader::reconstructSection(SmallVectorImpl<char> &SectionBuffer,
   /// copied the addend out of the Addends at a particular offset, we should
   /// skip all relocations that matches the same offset.
   int64_t PrevOffset = -1;
-  for (auto Reloc : Relocations.back()) {
-    auto RelocationOffsetInSection = getRelocationOffset(Reloc);
-    if (PrevOffset == RelocationOffsetInSection)
-      continue;
-    auto RelocationSize =
-        getRelocationSize(Reloc, getEndian() == endianness::little);
-    /// NumOfBytesToReloc: This denotes the number of bytes needed to be copied
-    /// into the \p SectionBuffer before we copy the next addend.
-    auto NumOfBytesToReloc = RelocationOffsetInSection - SectionBuffer.size();
-    // Copy the contents of the fragment till the next relocation.
-    SectionBuffer.append(FragmentBuffer.begin() + FragmentIndex,
-                         FragmentBuffer.begin() + FragmentIndex +
-                             NumOfBytesToReloc);
-    FragmentIndex += NumOfBytesToReloc;
-    // Copy the relocation addend.
-    SectionBuffer.append(Addends.begin() + AddendBufferIndex,
-                         Addends.begin() + AddendBufferIndex + RelocationSize);
-    AddendBufferIndex += RelocationSize;
-    PrevOffset = RelocationOffsetInSection;
+  /// If the \p Addends buffer is empty, there was no AddendsRef for this
+  /// section, this is either because no \p Relocations exist in this section,
+  /// or this is 32-bit architecture, where we do not support relocation
+  /// partitioning.
+  if (!Addends.empty()) {
+    for (auto Reloc : Relocations.back()) {
```

Comment:
It would have been cleaner if we could have made the `Relocations` vector empty, I'm guessing it is used somewhere else as well?

Reply:
We cannot do that, in the case where there are relocations in the section, but we didn't partition the section because it is 32 bit, we still have to write those relocations out into the object file.